core/DataTable/Filter/Sort.php

```diff
@@ -239,7 +232,7 @@ public function filter($table)
 
     private function getFirstValueFromDataTable($table)
     {
-        foreach ($table->getRows() as $row) {
+        foreach ($table->getRowsWithoutSummaryRow() as $row) {
```

Comment:
This is a 60 seconds performance improvement when archiving for instance a year period with a lot of data. Time spent here dropped from 60.2 seconds to 0.1 seconds.